pennylane/tape/qnode.py

```diff
@@ -447,6 +447,9 @@ def construct(self, args, kwargs):
                 stop_at=lambda obj: self.device.supports_operation(obj.name),
             )
 
+            # Check the observables without needing to create the circuit graph
+            self.qtape.is_sampled = any(obs.return_type == Sample for obs in self.qtape.observables)

```

Comment:
Really nice catch @antalszava! :100: 

I'm just wondering why the `is_sampled` method was changed to an attribute. Maybe this is nicer (and a bit more effective) to have it this way rather than always checking for `Sample` return types in the observables. :thinking:

Reply:
Hmm, so `is_sampled` is an attribute of `QuantumTape`, but a property of the `CircuitGraph`. Here at times, the circuit graph was not constructed, so simply checking the observables directly avoids the overhead of constructing the circuit graph

Reply:
I mean, an option would be to have `QuantumTape.is_sampled` a property instead of an attribute, as is the case for `CircuitGraph`. No need to create the graph in either case; it would be a small change. I'm not really advocating for either though, just a bit curious about the reasoning. :slightly_smiling_face:

Reply:
> Hmm, so is_sampled is an attribute of QuantumTape, but a property of the CircuitGraph. 

This was likely my fault, I didn't check that it was a property on the graph and just made it an attribute on the tape. However,

* this is probably not too much of an issue, since the tape is designed to be (mostly) non-mutable, whereas the circuit graph was very mutable.

* It was mostly a temporary fix, I needed the tape to 'look' like a CircuitGraph from a duck-typing perspective, so that existing devices would accept it. With the thinking that soon we would be able to drop this backward compatibility :)